controllers/common.go

```diff
@@ -44,6 +45,7 @@ func isNamespaceScoped(obj client.Object) bool {
 func BuildCollector(params manifests.Params) ([]client.Object, error) {
 	builders := []manifests.Builder{
 		collector.Build,
+		targetallocator.Build,
```

Comment:
I think the target allocator should be enabled/started only if the mode of the agent is set to stateful-set. From this line it seems it will always build the target-allocator

Reply:
For what I did here, the `targetallocator` only starts running when the `prometheusConfig` is proivded and `targetAllocator` is enabled in the `amazoncloudwatchagent` custom resource. This is likewise to upstream's functionality where it only starts when these are provided.

Regarding only starting when the agent is set to `statefulset`, that functionality should be added in a future PR.

Reply:
Yeah, I included this eventually since it should be enforced from the operator.